File: python/first-unique-character-in-a-string.py
from collections import Counter


class Solution:
  def firstUniqChar(self, s):
    """
    :type s: str
    :rtype: int
    """
    counts = Counter(s)
    for index, char in enumerate(s):
      if counts[char] == 1:
        return index
    return -1


s = Solution()
assert s.firstUniqChar('leetcode') == 0
assert s.firstUniqChar('loveleetcode') == 2

File: cpp/first-unique-character-in-a-string.cc
class Solution {
 public:
  int firstUniqChar(string s) {
    unordered_map<char, int> counts;
    for (char c : s) {
      if (counts.find(c) == counts.end()) {
        counts[c] = 1;
      } else {
        counts[c]++;
      }
    }

    for (int i = 0; i < s.length(); ++i) {
      if (counts[s[i]] == 1) {
        return i;
      }
    }
    return -1;
  }
};
